roles/common/tasks/config_validations.yml

```diff
@@ -238,17 +238,28 @@
 
 - assert:
     that:
-      - (kafka_controller_sasl_protocol | confluent.platform.split_to_list)[0] in ['plain','kerberos', 'oauth', 'none']
-    fail_msg: "kafka_controller_sasl_protocol supports ['plain', 'kerberos', 'oauth', 'none'] when providing single value."
+      - item in ['plain', 'scram', 'scram256', 'kerberos']
+    fail_msg: "kafka_controller_sasl_protocol value in CP-Ansible can have support for two values from the list ['plain', 'scram', 'scram256', 'kerberos']"
+  loop: "{{ kafka_controller_sasl_protocol | confluent.platform.split_to_list }}"
+  when: kafka_controller_sasl_protocol is defined and kafka_controller_sasl_protocol | confluent.platform.split_to_list | length == 2
+  tags: validate
+
+- assert:
+    that:
+      - (hostvars[item]['kafka_controller_sasl_protocol'] | confluent.platform.split_to_list)[0] not in ['scram', 'scram256']
+    fail_msg: "kafka_controller_sasl_protocol in kafka controller {{hostvars[item]['kafka_controller_sasl_protocol']}} cannot have first value as scram or scram256 since inter-controller communication via SCRAM is not supported. Please use other protocols along with it."
+  loop: "{{ groups['kafka_controller'] }}"
   when:
-    - kafka_controller_sasl_protocol is defined and kafka_controller_sasl_protocol | confluent.platform.split_to_list | length == 1
     - ('kafka_controller' in groups.keys() and groups['kafka_controller'] | length > 0) | bool
+    - hostvars[item]['kafka_controller_sasl_protocol'] is defined
   tags: validate
 
 - assert:
     that:
-      - item in ['plain', 'scram', 'scram256', 'kerberos']
-    fail_msg: "kafka_controller_sasl_protocol value in CP-Ansible can have support for two values from the list ['plain', 'scram', 'scram256', 'kerberos']"
-  loop: "{{ kafka_controller_sasl_protocol | confluent.platform.split_to_list }}"
-  when: kafka_controller_sasl_protocol is defined and kafka_controller_sasl_protocol | confluent.platform.split_to_list | length == 2
+      - hostvars[item]['kafka_controller_sasl_protocol'] is defined
+    fail_msg: "kafka_controller_sasl_protocol must be defined."
```

Comment:
is there any specific need for this check ? 
We are defining it to take the value form sasl_protocol which is by default none. so even if someone doesn't define this variable it would still be defined by variables role.

Reply:
This is making sure that if a customer runs validate first then it should prompt them to provide value of kafka_controller_sasl_protocol in case your sasl_protocol has first element as scram.